R/utils-shared.R

```diff
@@ -534,25 +534,37 @@ RescaleYTop <- function(object) {
   # graphics::par.
   m <- 0.04
   
+  # vertical extent and length
+  e <- ylim(object)$side.2
+  e.length <- abs(e[1] - e[2])
+  
   if (ylog) {
+    # TODO: the log10 calculations below should probably be upgraded to mirror
+    # the form of analogous formulae in the linear-case, conditionals in the
+    # block below.
+    
     # if the y-axis is inverted
     if (reverse) {
-      object$side.2$lim[1] <- 10^((1 - m) * log10(object$side.2$lim[1]))
+      object$side.2$lim[1] <- 10^((1 - m) * log10(e[2]))
     }
     else {
-      object$side.2$lim[2] <- 10^((1 + m) * log10(object$side.2$lim[2]))
+      object$side.2$lim[2] <- 10^((1 + m) * log10(e[2]))
     }
   }
   else {
     # if the y-axis is inverted
     if (reverse) {
-      object$side.2$lim[1] <- (1 - m) * object$side.2$lim[1]
+      object$side.2$lim[1] <- e[2] - m * e.length
     }
     else {
-      object$side.2$lim[2] <- (1 + m) * object$side.2$lim[2]
+      # The 5.14 coefficient below is a hack that was reverse-engineered from 
+      # the SVG output (using Inkscape, and the "back-of-the-envelope"). At the 
+      # moment, we have no idea why it works, and is likely not robust enough 
+      # for production.
+      object$side.2$lim[2] <- e[2] + 5.14 * m * e.length
```

Comment:
I think I might've been incorrect with that earlier assessment. I thought it was at the time; however, today I think it was affecting them both, but only when the _y_-axis was not inverted (why, I still don't know). I think earlier I assumed it was a DV-hydrograph-only problem, because the test data for DV hydrographs had mostly not-inverted _y_ axes, and the five year GW summary had an inverted _y_-axis.